### ISSUE TITLE
Implement healthcheck conditions used in compose

### ISSUE THREAD

**Issue by hellt** [srl-labs/containerlab#1421]
We have more and more sophisticated labs which introduce more and more external systems.
Quite often the external systems must depend one on another, for example, a database client needs the database to be healthy first.

Docker compose has a notion of health checks which might be defined in a compose file:

```yaml
  postgres:
    image: postgres:15.3
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -d irrd" ]
      interval: 15s
      timeout: 15s
      retries: 5
```

These kind of health-checks would be super useful to have in containerlab, and I suspect they might be integrated without substantial effort if we use the same syntax and call the relevant APIs.

@steiler What do you think?

**Comment by steiler** [comment]
Healthchecks might also be already be baked into the image, for general purpose healt checks.
https://docs.docker.com/engine/reference/builder/#healthcheck
It might be incarnation specific checks that we could support I assume.

**Comment by hellt** [comment]
Implemented in #1426, with the goal to add them to dep manager as per #1700
